### ISSUE TITLE
Unable to load simple image which was working before and it is not working now

### ISSUE THREAD

**Issue by SatishBala** [flutter/flutter#16127]
pubspec.yaml
```yaml
name: layout_app
description: A new Flutter project.

dependencies:
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.0

dev_dependencies:
  flutter_test:
    sdk: flutter


# For information on the generic Dart part of this file, see the
# following page: https://www.dartlang.org/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  assets:
    - images/download.jpeg
  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true
```

main.dart
```dart
           new Container(
                child: new Image.asset('download.jpeg')
            )
```

Logs
```
 flutter run
Running "flutter packages get" in layout_app...              0.5s
Using hardware rendering with device Android SDK built for x86. If you get graphics artifacts, consider enabling software rendering with "--enable-software-rendering".
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...                                       1.8s
Resolving dependencies...                                    0.7s
Running 'gradlew assembleDebug'...                           1.2s
Built build/app/outputs/apk/debug/app-debug.apk (24.8MB).
I/FlutterActivityDelegate( 5729): onResume setting current activity to this
Syncing files to device Android SDK built for x86...             
E/flutter ( 5729): [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter ( 5729): 'package:flutter/src/services/asset_bundle.dart': error: line 230: semicolon expected
E/flutter ( 5729):     0 throw new FlutterError("Unable to load asset: $key");
E/flutter ( 5729):      ^
E/flutter ( 5729): 
E/flutter ( 5729): #0      CachingAssetBundle._fetchString (package:flutter/src/services/asset_bundle.dart:160)
E/flutter ( 5729): <asynchronous suspension>
E/flutter ( 5729): #1      CachingAssetBundle.loadString (package:flutter/src/services/asset_bundle.dart:156)
E/flutter ( 5729): #2      CachingAssetBundle.loadStructuredData (package:flutter/src/services/asset_bundle.dart:193)
E/flutter ( 5729): #3      AssetImage.obtainKey (package:flutter/src/painting/image_resolution.dart:169)
E/flutter ( 5729): #4      ImageProvider.resolve (package:flutter/src/painting/image_provider.dart:264)
E/flutter ( 5729): #5      _ImageState._resolveImage (package:flutter/src/widgets/image.dart:526)
E/flutter ( 5729): #6      _ImageState.didChangeDependencies (package:flutter/src/widgets/image.dart:501)
E/flutter ( 5729): #7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3750)
E/flutter ( 5729): #8      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600)
E/flutter ( 5729): #9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #10     Element.updateChild (package:flutter/src/widgets/framework.dart:2693)
E/flutter ( 5729): #11     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636)
E/flutter ( 5729): #12     Element.rebuild (package:flutter/src/widgets/framework.dart:3478)
E/flutter ( 5729): #13     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605)
E/flutter ( 5729): #14     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600)
E/flutter ( 5729): #15     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #16     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4738)
E/flutter ( 5729): #17     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #18     Element.updateChild (package:flutter/src/widgets/framework.dart:2693)
E/flutter ( 5729): #19     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636)
E/flutter ( 5729): #20     Element.rebuild (package:flutter/src/widgets/framework.dart:3478)
E/flutter ( 5729): #21     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605)
E/flutter ( 5729): #22     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600)
E/flutter ( 5729): #23     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #24     Element.updateChild (package:flutter/src/widgets/framework.dart:2693)
E/flutter ( 5729): #25     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636)
E/flutter ( 5729): #26     Element.rebuild (package:flutter/src/widgets/framework.dart:3478)
E/flutter ( 5729): #27     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605)
E/flutter ( 5729): #28     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600)
E/flutter ( 5729): #29     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #30     Element.updateChild (package:flutter/src/widgets/framework.dart:2693)
E/flutter ( 5729): #31     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636)
```

```
shashwats-new-iMac:layout_app merudesigns$ flutter doctor -v
[✓] Flutter (Channel beta, v0.2.3, on Mac OS X 10.13.3 17D102, locale en-US)
    • Flutter version 0.2.3 at /Users/merudesigns/flutter
    • Framework revision 5a58b36e36 (3 weeks ago), 2018-03-13 13:20:13 -0700
    • Engine revision e61bb9ac3a
    • Dart version 2.0.0-dev.35.flutter-290c576264

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/merudesigns/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    • All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    ✗ ideviceinstaller is not installed; this is used to discover connected iOS devices.
      To install, run:
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.2
    • CocoaPods version 1.4.0

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)

! Doctor found issues in 1 category.
shashwats-new-iMac:layout_app merudesigns$ B
```

**Comment by SatishBala** [comment]
-- pubspec.yaml
name: layout_app
description: A new Flutter project.

dependencies:
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.0

dev_dependencies:
  flutter_test:
    sdk: flutter


# For information on the generic Dart part of this file, see the
# following page: https://www.dartlang.org/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  assets:
    - images/download.jpeg
  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true


main.dart

           new Container(
                child: new Image.asset('download.jpeg')
            )

Error
E/flutter ( 5178): [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter ( 5178): 'package:flutter/src/services/asset_bundle.dart': error: line 230 pos 8: semicolon expected
E/flutter ( 5178):       0throw new FlutterError('Unable to load asset: $key');
E/flutter ( 5178):        ^
E/flutter ( 5178): 
E/flutter ( 5178): #0      CachingAssetBundle._fetchString (package:flutter/src/services/asset_bundle.dart:160:33)
E/flutter ( 5178): <asynchronous suspension>
E/flutter ( 5178): #1      CachingAssetBundle.loadString (package:flutter/src/services/asset_bundle.dart:156:12)
E/flutter ( 5178): #2      CachingAssetBundle.loadStructuredData (package:flutter/src/services/asset_bundle.dart:193:5)
E/flutter ( 5178): #3      AssetImage.obtainKey (package:flutter/src/painting/image_resolution.dart:169:18)
E/flutter ( 5178): #4      ImageProvider.resolve (package:flutter/src/painting/image_provider.dart:264:5)
E/flutter ( 5178): #5      _ImageState._resolveImage (package:flutter/src/widgets/image.dart:526:20)
E/flutter ( 5178): #6      _ImageState.reassemble (package:flutter/src/widgets/image.dart:520:5)
E/flutter ( 5178): #7      StatefulElement._reassemble (package:flutter/src/widgets/framework.dart:3725:11)
E/flutter ( 5178): #8      Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #9      ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3659:14)
E/flutter ( 5178): #10     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #11     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #12     MultiChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4721:16)
E/flutter ( 5178): #13     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #14     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #15     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3659:14)
E/flutter ( 5178): #16     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #17     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #18     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3659:14)
E/flutter ( 5178): #19     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #20     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #21     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3659:14)
E/flutter ( 5178): #22     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #23     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #24     MultiChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4721:16)
E/flutter ( 5178): #25     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #26     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #27     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3659:14)
E/flutter ( 5178): #28     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #29     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #30     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3659:14)
E/flutter ( 5178): #31     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #32     StatefulElement._reassemble (package:flutter/src/widgets/framework.dart:3726:22)
E/flutter ( 5178): #33     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #34     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4621:14)
E/flutter ( 5178): #35     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #36     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #37     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3659:14)
E/flutter ( 5178): #38     Element._reassemble (package:flutter/src/widgets/framework.dart:2540:5)
E/flutter ( 5178): #39     Element._reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2541:13)
E/flutter ( 5178): #40     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4621:14)
E/flutter ( 5178): #41     Element.
Reloaded 0 of 483 libraries in 938ms.

flutter doctor -v

shashwats-new-iMac:layout_app merudesigns$ flutter doctor -v
[✓] Flutter (Channel beta, v0.2.3, on Mac OS X 10.13.3 17D102, locale en-US)
    • Flutter version 0.2.3 at /Users/merudesigns/flutter
    • Framework revision 5a58b36e36 (3 weeks ago), 2018-03-13 13:20:13 -0700
    • Engine revision e61bb9ac3a
    • Dart version 2.0.0-dev.35.flutter-290c576264

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/merudesigns/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    • All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    ✗ ideviceinstaller is not installed; this is used to discover connected iOS devices.
      To install, run:
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.2
    • CocoaPods version 1.4.0

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)

! Doctor found issues in 1 category.
shashwats-new-iMac:layout_app merudesigns$

**Comment by SatishBala** [comment]
Sorry for the messup.. first time posting

**Comment by chinmaygarde** [comment]
This issue is a bit hard to follow. Github allows you to edit posts however. Can you attach a reduced test case and logs as a gist?

**Comment by SatishBala** [comment]
Thanks, 

Context: I am trying to load an image and I am getting the following error consistently

My configuration

pubspec.yaml
-============
name: layout_app
description: A new Flutter project.

dependencies:
flutter:
sdk: flutter

flutter:

assets:
- images/download.jpeg


main.dart
==============
       new Container(
            child: new Image.asset('download.jpeg')
        )


ERRORS WHILE RUNNING "FLUTTER RUN"
Logs
flutter run
Running "flutter packages get" in layout_app... 0.5s
Using hardware rendering with device Android SDK built for x86. If you get graphics artifacts, consider enabling software rendering with "--enable-software-rendering".
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle... 1.8s
Resolving dependencies... 0.7s
Running 'gradlew assembleDebug'... 1.2s
Built build/app/outputs/apk/debug/app-debug.apk (24.8MB).
I/FlutterActivityDelegate( 5729): onResume setting current activity to this
Syncing files to device Android SDK built for x86...
E/flutter ( 5729): [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter ( 5729): 'package:flutter/src/services/asset_bundle.dart': error: line 230: semicolon expected
E/flutter ( 5729): 0 throw new FlutterError("Unable to load asset: $key");
E/flutter ( 5729): ^
E/flutter ( 5729):
E/flutter ( 5729): #0 CachingAssetBundle._fetchString (package:flutter/src/services/asset_bundle.dart:160)
E/flutter ( 5729): 
E/flutter ( 5729): #1 CachingAssetBundle.loadString (package:flutter/src/services/asset_bundle.dart:156)
E/flutter ( 5729): #2 CachingAssetBundle.loadStructuredData (package:flutter/src/services/asset_bundle.dart:193)
E/flutter ( 5729): #3 AssetImage.obtainKey (package:flutter/src/painting/image_resolution.dart:169)
E/flutter ( 5729): #4 ImageProvider.resolve (package:flutter/src/painting/image_provider.dart:264)
E/flutter ( 5729): #5 _ImageState._resolveImage (package:flutter/src/widgets/image.dart:526)
E/flutter ( 5729): #6 _ImageState.didChangeDependencies (package:flutter/src/widgets/image.dart:501)
E/flutter ( 5729): #7 StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3750)
E/flutter ( 5729): #8 ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600)
E/flutter ( 5729): #9 Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #10 Element.updateChild (package:flutter/src/widgets/framework.dart:2693)
E/flutter ( 5729): #11 ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636)
E/flutter ( 5729): #12 Element.rebuild (package:flutter/src/widgets/framework.dart:3478)
E/flutter ( 5729): #13 ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605)
E/flutter ( 5729): #14 ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600)
E/flutter ( 5729): #15 Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #16 MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4738)
E/flutter ( 5729): #17 Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #18 Element.updateChild (package:flutter/src/widgets/framework.dart:2693)
E/flutter ( 5729): #19 ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636)
E/flutter ( 5729): #20 Element.rebuild (package:flutter/src/widgets/framework.dart:3478)
E/flutter ( 5729): #21 ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605)
E/flutter ( 5729): #22 ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600)
E/flutter ( 5729): #23 Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #24 Element.updateChild (package:flutter/src/widgets/framework.dart:2693)
E/flutter ( 5729): #25 ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636)
E/flutter ( 5729): #26 Element.rebuild (package:flutter/src/widgets/framework.dart:3478)
E/flutter ( 5729): #27 ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605)
E/flutter ( 5729): #28 ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600)
E/flutter ( 5729): #29 Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890)
E/flutter ( 5729): #30 Element.updateChild (package:flutter/src/widgets/framework.dart:2693)
E/flutter ( 5729): #31 ComponentElement.performRebuild 

FLUTTER DOCTOR OUTPUT

flutter doctor -v

shashwats-new-iMac:layout_app merudesigns$ flutter doctor -v
[✓] Flutter (Channel beta, v0.2.3, on Mac OS X 10.13.3 17D102, locale en-US)
• Flutter version 0.2.3 at /Users/merudesigns/flutter
• Framework revision 5a58b36 (3 weeks ago), 2018-03-13 13:20:13 -0700
• Engine revision e61bb9ac3a
• Dart version 2.0.0-dev.35.flutter-290c576264

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
• Android SDK at /Users/merudesigns/Library/Android/sdk
• Android NDK location not configured (optional; useful for native profiling support)
• Platform android-27, build-tools 27.0.3
• Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
• Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
• All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 9.2)
• Xcode at /Applications/Xcode.app/Contents/Developer
• Xcode 9.2, Build version 9C40b
✗ ideviceinstaller is not installed; this is used to discover connected iOS devices.
To install, run:
brew install --HEAD libimobiledevice
brew install ideviceinstaller
• ios-deploy 1.9.2
• CocoaPods version 1.4.0

[✓] Android Studio (version 3.0)
• Android Studio at /Applications/Android Studio.app/Contents
• Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] Connected devices (1 available)
• Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)

! Doctor found issues in 1 category.

**Comment by jason-simmons** [comment]
The `0` in the exception log is suspicious:

```
E/flutter ( 5729): 'package:flutter/src/services/asset_bundle.dart': error: line 230: semicolon expected
E/flutter ( 5729): 0 throw new FlutterError("Unable to load asset: $key");
E/flutter ( 5729): ^
E/flutter ( 5729):
E/flutter ( 5729): #0 CachingAssetBundle._fetchString (package:flutter/src/services/asset_bundle.dart:160)
```

If you go to the Flutter framework source code in your repository and open `packages/flutter/lib/src/services/asset_bundle.dart` then do you see the `0` character on line 160?

That character should not be there. If it's there then your Flutter repository may be in a corrupt state.

**Comment by Hixie** [comment]
@SatishBala We think this was probably caused by you accidentally editing your flutter repository, so I'm going to close this bug. If you still see this please file a new bug with as much information as you can provide. Thanks!
